### ISSUE TITLE
TimeSeries API PyTorch Engine support

### ISSUE THREAD

**Issue by keklol5050** [deepjavalibrary/djl#3259]
The documentation states (https://djl.ai/extensions/timeseries/docs/forecast_with_M5_data.html) that PyTorch also supports the TimeSeries API, but when trying to train a model with the PyTorch engine, an error occurs:
Exception in thread "main" java.lang.UnsupportedOperationException: Not implemented yet.
	at ai.djl.pytorch.engine.PtNDArray.gammaln(PtNDArray.java:845)
	at ai.djl.timeseries.distribution.NegativeBinomial.logProb(NegativeBinomial.java:47)
	at ai.djl.timeseries.distribution.DistributionLoss.evaluate(DistributionLoss.java:54)
	at ai.djl.training.EasyTrain.trainSplit(EasyTrain.java:124)
	at ai.djl.training.EasyTrain.trainBatch(EasyTrain.java:110)
	at ai.djl.training.EasyTrain.fit(EasyTrain.java:58)
	at com.crypto.analysis.main.ex.NNew.train(NNew.java:134)
	at com.crypto.analysis.main.ex.NNew.main(NNew.java:63)

`public static void train(Path outputDir) {
        try (Model model = Model.newInstance("DeepAR");
             NDManager manager = NDManager.newBaseManager()) {

            DistributionOutput distributionOutput = new NegativeBinomialOutput();
            DefaultTrainingConfig config = setupTrainingConfig(distributionOutput);
            DeepARNetwork trainingNetwork = getDeepARModel(freq, predictionLength, distributionOutput, true);

            model.setBlock(trainingNetwork);

            List<TimeSeriesTransform> trainingTransformation = trainingNetwork.createTrainingTransformation(manager);
            int contextLength = trainingNetwork.getContextLength();

            NDataset trainSet = getDataset(trainingTransformation, contextLength);

            try (Trainer trainer = model.newTrainer(config)) {
                trainer.setMetrics(new Metrics());

                System.out.println("+++++" + trainSet.availableSize());
                int historyLength = trainingNetwork.getHistoryLength();
                Shape[] inputShapes = new Shape[4];
                // (N, num_cardinality)
                inputShapes[0] = new Shape(1, 1);
                // (N, num_real) if use_feat_stat_real else (N, 1)
                inputShapes[1] = new Shape(1, 1);

                inputShapes[2] =
                        new Shape(
                                1,
                                historyLength,
                                TimeFeature.timeFeaturesFromFreqStr("M").size() + 1);
                inputShapes[3] = new Shape(1, historyLength);
                trainer.initialize(inputShapes);

                EasyTrain.fit(trainer, 1500, trainSet, null);
                logger.info(String.valueOf(trainer.getTrainingResult()));

                model.setProperty("Epochs", "1500");
                model.save(outputDir, "DeepAR");
            }

        } catch (TranslateException | IOException e) {
            throw new RuntimeException(e);
        }
    }

    private static DeepARNetwork getDeepARModel(String freq, int predictionLength,
                                                DistributionOutput distributionOutput, boolean training) {

        List<Integer> cardinality = new ArrayList<>();
        cardinality.add(allSize-testSize);

        DeepARNetwork.Builder builder =
                DeepARNetwork.builder()
                        .setFreq(freq)
                        .setPredictionLength(predictionLength)
                        .setCardinality(cardinality)
                        .optDistrOutput(distributionOutput)
                        .optUseFeatDynamicReal(false)
                        .optUseFeatStaticCat(false)
                        .optUseFeatStaticReal(false);
        return training ? builder.buildTrainingNetwork() : builder.buildPredictionNetwork();
    }

    private static DefaultTrainingConfig setupTrainingConfig(DistributionOutput distributionOutput) {
        return new DefaultTrainingConfig(new DistributionLoss("Loss", distributionOutput))
                .addEvaluator(new Rmsse(distributionOutput))
                .optInitializer(new XavierInitializer(), Parameter.Type.WEIGHT)
                .addTrainingListeners(TrainingListener.Defaults.logging());
    }`
